lib/src/components/NxIconDropdown/__tests__/NxIconDropdown.test.tsx

```diff
@@ -5,326 +5,298 @@
+
+    expect(toggleBtn).toBeInTheDocument();
+
+    await runTimers();
+
+    expect(toggleBtn).toHaveAccessibleName('Test tooltip');
   });
 
-  it('renders the children within the NxDropdownMenu, wrapping nx-dropdown-links and nx-dropdown-buttons ' +
-      'in NxOveflowTooltip', function() {
+  it('renders the children within the dropdown menu in the specified order', function() {
     const children = [
-      <a id="link1" key="1">Link1</a>,
-      <a id="link2" className="nx-dropdown-link" key="2">Link2</a>,
-      <button id="link3" className="nx-dropdown-button" key="3">Link3</button>,
-      <button id="link4" className="nx-dropdown-right-button" key="4">Link4</button>
+      <a data-testid="menu-child" key="1">Link1</a>,
+      <a data-testid="menu-child" key="2">Link2</a>,
+      <button data-testid="menu-child" key="3">Link3</button>,
+      <button data-testid="menu-child" key="4">Link4</button>
     ];
-    const component = getMountedComponent({ children, isOpen: true }),
-        menu = component.find('div.nx-dropdown-menu');
 
-    expect(menu.childAt(0)).toMatchElement(<a id="link1" key="1">Link1</a>);
+    const el = renderEl({ children, isOpen: true })!,
+        menu = el.querySelector('.nx-dropdown-menu'),
+        menuChildren = within(el).getAllByTestId('menu-child');
 
-    expect(menu.childAt(1)).toMatchSelector(NxOverflowTooltip);
-    expect(menu.childAt(1).find('a'))
-        .toMatchElement(<a id="link2" key="2" className="nx-dropdown-link">Link2</a>);
-
-    expect(menu.childAt(2)).toMatchSelector(NxOverflowTooltip);
-    expect(menu.childAt(2).find('button')).toMatchElement(
-      <button id="link3" key="3" className="nx-dropdown-button">Link3</button>
-    );
-
-    expect(menu.childAt(3).find('button')).toMatchElement(
-      <button id="link4" key="4" className="nx-dropdown-right-button">Link4</button>
-    );
+    expect(menu).toBeInTheDocument();
+    expect(menuChildren[0]).toHaveTextContent('Link1');
+    expect(menuChildren[1]).toHaveTextContent('Link2');
+    expect(menuChildren[2]).toHaveTextContent('Link3');
+    expect(menuChildren[3]).toHaveTextContent('Link4');
   });
 
-  it('calls onToggleCollapse if a click happens anywhere when the dropdown is already open', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse, isOpen: true }, { attachTo: container });
+  it('calls onToggleCollapse if a click happens anywhere when the dropdown is already open', async function() {
+    const user = userEvent.setup(),
+        onToggleCollapse = jest.fn();
+
+    quickRender({ onToggleCollapse, isOpen: true });
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
+
     expect(onToggleCollapse).toHaveBeenCalled();
   });
```

Comment:
On thing I realize we're missing, here and probably also in `NxDropdown`, is a test that clicking the items within the dropdown 1. fires their click handlers, and 2. closes the dropdown.

Reply:
> probably also in NxDropdown

Added a note in the RSC-989 ticket to address this.



> a test that clicking the items within the dropdown 1. fires their click handlers, and 2. fires onToggleCollapse

Added here 654d52f7b11ba5fd3adbdb43c4f6607757d03843